src/main/java/org/generationcp/middleware/dao/SampleDao.java

```diff
@@ -440,6 +450,40 @@ public long countSampleObservationDtos(final SampleSearchRequestDTO requestDTO)
 		return ((BigInteger) sqlQuery.uniqueResult()).longValue();
 	}
 
+	public void deleteByListAndEntryIds(final Integer sampleListId, final List<Integer> sampleIds) {
+		Preconditions.checkNotNull(sampleListId, "sampleListId can not be null.");
```

Comment:
Is sampleListId necessary here or sampleIds is enough?

Reply:
Hi @abatac , I think it's necessary so that the samples that will be deleted are in the same sample list and the entry numbers  will be reordered correctly.